lib/teleterm/clusters/storage.go

```diff
@@ -202,10 +204,15 @@ func (s *Storage) fromProfile(clusterName string) (*Cluster, error) {
 		return nil, trace.Wrap(err)
 	}
 
-	clusterURI := uri.NewClusterURI(clusterName)
+	clusterURI := uri.NewClusterURI(profileName)
+	if leafClusterName != "" {
+		clusterURI = clusterURI.AppendLeafCluster(leafClusterName)
+	}
```

Comment:
Previously this was done in `Storage.GetByURI`, but I see no reason why we couldn't do it here.